fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/IntersectionSituationDataBuilder.java

```diff
@@ -308,6 +316,43 @@ public BaseLayer buildBaseLayer(IntersectionInputData isdInputData) {
 		return baseLayer;
 	}
 
+	/**
+	 * This function builds and returns list of geometry containers for RGA Data
+	 */
+	public List<GeometryContainer> buildGeometryContainers(IntersectionInputData isdInputData) {
+		List<GeometryContainer> geometryContainers = new ArrayList<>();
+
+		// Checking if approaches are null 
+		if (isdInputData.mapData.intersectionGeometry.laneList.approach == null) {
+			return geometryContainers;
+		}
+
+		Approach[] approaches = isdInputData.mapData.intersectionGeometry.laneList.approach;
+		GeometryContainer geometryContainer = new GeometryContainer();
+		ApproachGeometryLayer approachGeometryLayer = new ApproachGeometryLayer();
+
+		for (int approachIndex = 0; approachIndex < approaches.length; approachIndex++) {
+			Approach approach = approaches[approachIndex];
+
+			// Omitting crosswalk lanes
```

Comment:
I think better to say "excluding crosswalk lanes"
also add a line saying crosswalks dont have approach id and it is default to -1